File: JavaScript/Practice/LeetCode/Easy/Determine if String Halves Are Alike.js
// https://leetcode.com/problems/determine-if-string-halves-are-alike/description/

/**
 * @param {string} s
 * @return {boolean}
 */
var halvesAreAlike = function(s) {
    const firstHalf = [];
    const secondHalf = [];
    const vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];

    for (let i = 0; i < s.length / 2; i++) {
        let firstHalfLetter = s[i];
        let secondHalfLetter = s[s.length - i - 1];
        if (vowels.includes(firstHalfLetter)) {
            firstHalf.push(firstHalfLetter);
        }
        if (vowels.includes(secondHalfLetter)) {
            secondHalf.push(secondHalfLetter);
        }
    }
return firstHalf.length === secondHalf.length;    
};

File: C#/Practice/LeetCode/Easy/Determine if String Halves Are Alike.cs
// https://leetcode.com/problems/determine-if-string-halves-are-alike/description/

public class Solution {
    public bool HalvesAreAlike(string s) {
        List<char> firstHalf = new List<char>();
        List<char> secondHalf = new List<char>();
        char[] vowels = { 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U' };

        for (int i = 0; i < s.Length / 2; i++)
        {
            char firstHalfLetter = s[i];
            char secondHalfLetter = s[s.Length - i - 1];

            if (Array.Exists(vowels, v => v == firstHalfLetter))
            {
                firstHalf.Add(firstHalfLetter);
            }

            if (Array.Exists(vowels, v => v == secondHalfLetter))
            {
                secondHalf.Add(secondHalfLetter);
            }
        }

        return firstHalf.Count == secondHalf.Count;
    }

}